examples/base/solver_tutorial.py

```diff
@@ -37,40 +37,61 @@
 
 # %%
 import enum
-from typing import Any, Dict
+from typing import Dict
 
 import matplotlib.pyplot as plt
 import numpy as np
 from scipy.optimize import curve_fit
 
+from bluemira.base.parameter import ParameterFrame
 from bluemira.base.solver import RunMode, SolverABC, Task
 
 # %%[markdown]
 # Let's generate some noisy (Gaussian) data to use in the example.
 
 
 # %%
-def gaussian(x, sigma, mu, vertical_offset):
+def gaussian(x: np.ndarray, sigma: float, mu: float, vertical_offset: float):
     """Apply the Gaussian function to x."""
     exponent = -1 / 2 * ((x - mu) / sigma) ** 2
     return 1 / (sigma * np.sqrt(2 * np.pi)) * np.exp(exponent) + vertical_offset
 
 
-def generate_noisy_gaussian(x: np.ndarray, gauss_params: Dict[str, float], rng_seed=0):
+def generate_noisy_gaussian(
+    x: np.ndarray, sigma: float, mu: float, vertical_offset: float, rng_seed: int = 0
+):
     """Generate some gaussian data with random noise."""
-    gauss_y = gaussian(x, **gauss_params)
+    gauss_y = gaussian(x, sigma, mu, vertical_offset)
     rng = np.random.default_rng(rng_seed)
     return gauss_y + rng.uniform(-0.05, 0.05, len(x))
 
 
 # %%
 x = np.linspace(-10, 10, 200)
-gauss_params = {
-    "sigma": 2,
-    "mu": 1,
-    "vertical_offset": 1.5,
-}
-gauss_data = generate_noisy_gaussian(x, gauss_params)
+gauss_data = generate_noisy_gaussian(x, 2, 1.8, 1.5)
+gauss_params = ParameterFrame(
+    [
+        ["sigma", "sigma", 1.0, "dimensionless", "Standard Deviation", "Input"],
+        ["mu", "mu", 1.0, "dimensionless", "Expectation value", "Input"],
+        [
+            "vertical_offset",
+            "vertical_offset",
+            0.0,
+            "dimensionless",
+            "The vertical offset",
+            "Input",
+        ],
+        ["x", "x", x, "dimensionless", "Problem space", "Input"],
+        [
+            "gauss_data",
+            "gauss_data",
+            gauss_data,
+            "dimensionless",
+            "Gauss signal",
+            "Input",
+        ],
```

Comment:
I know what you mean about it feeling hacky but I think its fine. `Parameter` was semi designed for arrays anyway we just dont use it for them generally